### DISCUSSION TITLE
Why bash script run default user root. can change to another user to run bash scripts?

### DISCUSSION THREAD

**Discussion by yasok787** [semaphoreui/semaphore#2139] [Q&A]
### Question

9:38:28 AM
Task 103 added to queue
9:38:30 AM
Started: 103
9:38:30 AM
Run TaskRunner with template: repmgr
9:38:30 AM
Preparing: 103
9:38:31 AM
db-postgres | UNREACHABLE! => {
9:38:31 AM
    "changed": false,
9:38:31 AM
    "msg": "Failed to connect to the host via ssh: root@192.168.1.1: Permission denied (publickey,gssapi-keyex,gssapi-with-mic,password).",
9:38:31 AM
    "unreachable": true
9:38:31 AM
}
9:38:31 AM
Running playbook failed: exit status 4

**Comment by fiftin** [comment]
Need more details: platform, version, logs, etc.

**Reply by yasok787** [reply]
I have installed Semaphore UI from Binary file, and then created task bash scripts but when we run the task. defaut user is: root that login connect to  host to run shell script. how can we change to anthor user to run bash scripts.

##Resutl log task
9:54:44 AM
Task 118 added to queue
9:54:45 AM
Started: 118
9:54:45 AM
Run TaskRunner with template: whoami.sh
9:54:45 AM
Preparing: 118
9:54:45 AM
root


##Here's my shell scripts.
[postgres@test01]$ more whoami.sh 
whoami

**Reply by fiftin** [reply]
Are you started Semaphore UI under root user?

**Reply by yasok787** [reply]
yes sir.

**Reply by fiftin** [reply]
@yasok787 currently Semaphore runs Ansible/Terraform/Bash under the same user. But I agree it is good idea to allow use different user ;)
